pkgs/development/compilers/ghc/common-hadrian.nix

```
@@ -385,6 +385,10 @@ stdenv.mkDerivation ({
     "--enable-dwarf-unwind"
     "--with-libdw-includes=${lib.getDev elfutils}/include"
     "--with-libdw-libraries=${lib.getLib elfutils}/lib"
+  ] ++ lib.optionals targetPlatform.isDarwin [
```

Do you think it makes sense to make this a bit more specific?

```suggestion
  ] ++ lib.optionals (targetCC.bintools.isLLVM && targetPlatform.isDarwin) [
```

---

Darwin’s bintools are a mix of LLVM and cctools. Which is which also depends on the LLVM version. I’ll have to double check, but I’m pretty sure they do not set `isLLVM` because they’re not purely LLVM tools (for the same reason the stdenv does not).

---

Okay, makes sense. The build should succeed as well with pure cctools I'm guessing? If so, everything's fine.

---

Correct. The change is forcing `configure` to ignore that `ar` supports `-L`. That matches how it handles cctools `ar`, which does not support `-L` at all.

---

I double checked. My recollection was accurate. cctools-llvm is not setting `isLLVM`.